src/runtime/sgx/trusted/ecall_registry.cc

```diff
@@ -0,0 +1,109 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file ecall_registry.cc
+ * \brief The global registry of packed functions available via ecall_packed_func.
```

Comment:
Put this as a header file, and add def guard, so it also works under non-amalgamation mode if we decided to do so later